File: docs/snippets/python/multimodal-message_main.py
# from microsoft.agents.xml import MessageSerializer
# from microsoft.agents.xml.models import ChatMessage, TextContent, ImageContent
#
# # Create a message with text and image
# message = ChatMessage(
#     role="user",
#     message_id="msg-002",
#     contents=[
#         TextContent(text="What's in this image?"),
#         ImageContent(
#             uri="https://example.com/image.jpg",
#             alt_text="A photo of a sunset"
#         )
#     ]
# )
#
# # Serialize to XML
# serializer = MessageSerializer()
# xml_output = serializer.serialize(message)

File: snippets/csharp/multimodal-message_main.cs
// using Microsoft.Agents.Xml.Generated.Models;
// using Microsoft.Agents.Protocol.Xml;

// Create a message with text and image
var message = new UserMessage
{
    MessageId = "msg-002",
    Contents = new List<AIContent>
    {
        new TextContent { Text = "What's in this image?" },
        new ImageContent
        {
            Uri = "https://example.com/image.jpg",
            Alt = "A photo of a sunset"
        }
    }
};

// Serialize to XML
var serializer = new MessageSerializer();
var xmlOutput = serializer.Serialize(message);